server/app/auth/oidc/CiviformOidcProfileCreator.java

```diff
@@ -114,16 +114,17 @@ protected CiviFormProfileData mergeCiviFormProfile(
         .forEach(role -> civiformProfile.getProfileData().addRole(role));
     adaptForRole(civiformProfile, roles);
 
+    String emailAddress =
+        getEmail(oidcProfile)
+            .orElseThrow(
+                () -> new InvalidOidcProfileException("Unable to get email from profile."));
+
     // If the civiformProfile is a trusted intermediary, bypass remaining merging because
     // we don't want to actually merge the guest profile into theirs.
     if (isTrustedIntermediary(civiformProfile)) {
-      return civiformProfile.getProfileData();
+      return civiformProfile.getProfileData().setEmail(emailAddress);
```

Comment:
**suggestion (non-blocking):** Add a comment that explains why you are setting the email address here.

Reply:
Done